### DISCUSSION TITLE
tailwind css not working properly

### DISCUSSION THREAD

**Discussion by princepalewar12** [tailwindlabs/tailwindcss#12751] [Help]
Tailwind` is not working properly it is not adding css classes to my react project until i do some changes in tailwind.config.js file, changes like pressing a space or pressing a enter key. then all of a sudden i see the classes  has been added to the file in react.

**Comment by alex-krasikau** [comment]
Hey @princepalewar12!

Can you give me more info about your project setup? What framework are you using? Is it Create React App, Vite, Next.js, or Remix? Also, let me know the version of React.js and Tailwind CSS you're using, as well as your Tailwind CSS and Post CSS configs.

Once I have all the details, I'll be able to assist you with this problem.

**Reply by princepalewar12** [reply]
happy to hear from you alex . I would like to tell you more about my project and the issue I am facing. so I am developing a food delivery app in React js and I am using parcel bundler in it, i also installed tailwindcss using parcel from the documentation of tailwindcss. currently i am using react.js version 18.2.0. and tailwind CSS version 3.4.1.

here is my package.json file:
`{
  "name": "food-app",
  "version": "1.0.0",
  "description": "Food delivery App.",
  "scripts": {
    "start": " parcel index.html",
    "build": "parcel build index.html",
    "test": "jest"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/princepalewar12/Food-App.git"
  },
  "keywords": [
    "react",
    "javascript"
  ],
  "author": "Prince Palewar",
  "license": "ISC",
  "bugs": {
    "url": "https://github.com/princepalewar12/Food-App/issues"
  },
  "homepage": "https://github.com/princepalewar12/Food-App#readme",
    "devDependencies": {
      "assert": "^2.1.0",
      "buffer": "^6.0.3",
      "parcel": "^2.10.3",
      "postcss": "^8.4.33",
      "process": "^0.11.10",
      "tailwindcss": "^3.4.1"
    },
    "dependencies": {
      "react": "^18.2.0",
      "react-dom": "^18.2.0",
      "react-router-dom": "^6.21.1",
      "react-scripts": "^5.0.1"
    },
  "browserslist": [
    "last 2 versions"
  ]
}
`
here is the tailwind.config.js file:
`/** @type {import('tailwindcss').Config} */
module.exports = {
  content: ["./src/**/*.{html,js}"],
  theme: { 
    extend: {},
  },
  plugins: [],
};`

and here is the postcss file:
`{
  "plugins": {
    "tailwindcss": {}
  }
}`

**Reply by alex-krasikau** [reply]
This is definitely the parcel watch issue.

First, I really suggest you move your application to one of these frameworks: vite, next.js, remix. The react team recommends them for building things - https://react.dev/learn/start-a-new-react-project#production-grade-react-frameworks. That should fix the issue right away.

But if you really wanna stick with Parcel, can you please share your parcel configuration and the development script you use?

**Reply by princepalewar12** [reply]
### parcel configuration:
`{
  "name": "@parcel/config-default",
  "version": "2.10.3",
  "license": "MIT",
  "publishConfig": {
    "access": "public"
  },
  "funding": {
    "type": "opencollective",
    "url": "https://opencollective.com/parcel"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/parcel-bundler/parcel.git"
  },
  "main": "index.json",
  "scripts": {
    "test-ci": "mocha"
  },
  "dependencies": {
    "@parcel/bundler-default": "2.10.3",
    "@parcel/compressor-raw": "2.10.3",
    "@parcel/namer-default": "2.10.3",
    "@parcel/optimizer-css": "2.10.3",
    "@parcel/optimizer-htmlnano": "2.10.3",
    "@parcel/optimizer-image": "2.10.3",
    "@parcel/optimizer-svgo": "2.10.3",
    "@parcel/optimizer-swc": "2.10.3",
    "@parcel/packager-css": "2.10.3",
    "@parcel/packager-html": "2.10.3",
    "@parcel/packager-js": "2.10.3",
    "@parcel/packager-raw": "2.10.3",
    "@parcel/packager-svg": "2.10.3",
    "@parcel/packager-wasm": "2.10.3",
    "@parcel/reporter-dev-server": "2.10.3",
    "@parcel/resolver-default": "2.10.3",
    "@parcel/runtime-browser-hmr": "2.10.3",
    "@parcel/runtime-js": "2.10.3",
    "@parcel/runtime-react-refresh": "2.10.3",
    "@parcel/runtime-service-worker": "2.10.3",
    "@parcel/transformer-babel": "2.10.3",
    "@parcel/transformer-css": "2.10.3",
    "@parcel/transformer-html": "2.10.3",
    "@parcel/transformer-image": "2.10.3",
    "@parcel/transformer-js": "2.10.3",
    "@parcel/transformer-json": "2.10.3",
    "@parcel/transformer-postcss": "2.10.3",
    "@parcel/transformer-posthtml": "2.10.3",
    "@parcel/transformer-raw": "2.10.3",
    "@parcel/transformer-react-refresh-wrap": "2.10.3",
    "@parcel/transformer-svg": "2.10.3"
  },
  "parcelDependencies": {
    "@parcel/optimizer-data-url": "2.10.3",
    "@parcel/packager-raw-url": "2.10.3",
    "@parcel/packager-ts": "2.10.3",
    "@parcel/packager-xml": "2.10.3",
    "@parcel/transformer-coffeescript": "2.10.3",
    "@parcel/transformer-elm": "2.10.3",
    "@parcel/transformer-glsl": "2.10.3",
    "@parcel/transformer-graphql": "2.10.3",
    "@parcel/transformer-inline-string": "2.10.3",
    "@parcel/transformer-jsonld": "2.10.3",
    "@parcel/transformer-less": "2.10.3",
    "@parcel/transformer-mdx": "2.10.3",
    "@parcel/transformer-pug": "2.10.3",
    "@parcel/transformer-sass": "2.10.3",
    "@parcel/transformer-stylus": "2.10.3",
    "@parcel/transformer-sugarss": "2.10.3",
    "@parcel/transformer-toml": "2.10.3",
    "@parcel/transformer-typescript-types": "2.10.3",
    "@parcel/transformer-vue": "2.10.3",
    "@parcel/transformer-webmanifest": "2.10.3",
    "@parcel/transformer-worklet": "2.10.3",
    "@parcel/transformer-xml": "2.10.3",
    "@parcel/transformer-yaml": "2.10.3"
  },
  "peerDependencies": {
    "@parcel/core": "^2.10.3"
  },
  "gitHead": "65d42a955db665a04817fa9be55df16f588593d4"
}
`
### index.json of parcel:
`{
  "name": "@parcel/config-default",
  "version": "2.10.3",
  "license": "MIT",
  "publishConfig": {
    "access": "public"
  },
  "funding": {
    "type": "opencollective",
    "url": "https://opencollective.com/parcel"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/parcel-bundler/parcel.git"
  },
  "main": "index.json",
  "scripts": {
    "test-ci": "mocha"
  },
  "dependencies": {
    "@parcel/bundler-default": "2.10.3",
    "@parcel/compressor-raw": "2.10.3",
    "@parcel/namer-default": "2.10.3",
    "@parcel/optimizer-css": "2.10.3",
    "@parcel/optimizer-htmlnano": "2.10.3",
    "@parcel/optimizer-image": "2.10.3",
    "@parcel/optimizer-svgo": "2.10.3",
    "@parcel/optimizer-swc": "2.10.3",
    "@parcel/packager-css": "2.10.3",
    "@parcel/packager-html": "2.10.3",
    "@parcel/packager-js": "2.10.3",
    "@parcel/packager-raw": "2.10.3",
    "@parcel/packager-svg": "2.10.3",
    "@parcel/packager-wasm": "2.10.3",
    "@parcel/reporter-dev-server": "2.10.3",
    "@parcel/resolver-default": "2.10.3",
    "@parcel/runtime-browser-hmr": "2.10.3",
    "@parcel/runtime-js": "2.10.3",
    "@parcel/runtime-react-refresh": "2.10.3",
    "@parcel/runtime-service-worker": "2.10.3",
    "@parcel/transformer-babel": "2.10.3",
    "@parcel/transformer-css": "2.10.3",
    "@parcel/transformer-html": "2.10.3",
    "@parcel/transformer-image": "2.10.3",
    "@parcel/transformer-js": "2.10.3",
    "@parcel/transformer-json": "2.10.3",
    "@parcel/transformer-postcss": "2.10.3",
    "@parcel/transformer-posthtml": "2.10.3",
    "@parcel/transformer-raw": "2.10.3",
    "@parcel/transformer-react-refresh-wrap": "2.10.3",
    "@parcel/transformer-svg": "2.10.3"
  },
  "parcelDependencies": {
    "@parcel/optimizer-data-url": "2.10.3",
    "@parcel/packager-raw-url": "2.10.3",
    "@parcel/packager-ts": "2.10.3",
    "@parcel/packager-xml": "2.10.3",
    "@parcel/transformer-coffeescript": "2.10.3",
    "@parcel/transformer-elm": "2.10.3",
    "@parcel/transformer-glsl": "2.10.3",
    "@parcel/transformer-graphql": "2.10.3",
    "@parcel/transformer-inline-string": "2.10.3",
    "@parcel/transformer-jsonld": "2.10.3",
    "@parcel/transformer-less": "2.10.3",
    "@parcel/transformer-mdx": "2.10.3",
    "@parcel/transformer-pug": "2.10.3",
    "@parcel/transformer-sass": "2.10.3",
    "@parcel/transformer-stylus": "2.10.3",
    "@parcel/transformer-sugarss": "2.10.3",
    "@parcel/transformer-toml": "2.10.3",
    "@parcel/transformer-typescript-types": "2.10.3",
    "@parcel/transformer-vue": "2.10.3",
    "@parcel/transformer-webmanifest": "2.10.3",
    "@parcel/transformer-worklet": "2.10.3",
    "@parcel/transformer-xml": "2.10.3",
    "@parcel/transformer-yaml": "2.10.3"
  },
  "peerDependencies": {
    "@parcel/core": "^2.10.3"
  },
  "gitHead": "65d42a955db665a04817fa9be55df16f588593d4"
}
`
### Here is the development script of body compnent in my project:
`import { useEffect, useState } from "react";
import RestaurantCard from "./RestaurantCard";
import Shimmer from "./shimmer";
import { Link } from "react-router-dom";
import useOnlineStatus from "../ulits/useOnlineStatus";
import { withPromotedLabel } from "./RestaurantCard";

const Body = () => {
  const [listOfRestaurants, setListOfRestaurants] = useState([]);
  const [filteredRestaurants, setFilteredRestaurants] = useState([]);
  const [topRestaurants, setTopRestaurants] = useState([]);

  const [searchText, setSearchText] = useState("");

  const RestaurantCardPromoted = withPromotedLabel(RestaurantCard);




  useEffect(() => {
    fetchData();
  }, []);
  const fetchData = async () => {
    const data = await fetch(
      "https://www.swiggy.com/dapi/restaurants/list/v5?lat=30.3164945&lng=78.03219179999999&is-seo-homepage-enabled=true&page_type=DESKTOP_WEB_LISTING"
    );
    const json = await data.json();

    setListOfRestaurants(
      json?.data?.cards[1]?.card?.card?.gridElements?.infoWithStyle?.restaurants
    );
    setFilteredRestaurants(
      json?.data?.cards[1]?.card?.card?.gridElements?.infoWithStyle?.restaurants
    );
    setTopRestaurants(
      json?.data?.cards[1]?.card?.card?.gridElements?.infoWithStyle?.restaurants
    );
  };

  const onlineStatus = useOnlineStatus();

  if (onlineStatus === false)
    return <h1>Looks like you are Offline!! Please check your internet</h1>;

 
  return listOfRestaurants.length === 0 ? (
    <Shimmer />
  ) : (
    <div className="body">
      <div className="flex">
        <div className="m-4 p-4 flex item-center justify-center ">
          <button
            className="px-4 py-2 bg-gray-100 rounded-lg border border-emerald-950 "
            onClick={() => {
              const filteredList = topRestaurants.filter(
                (res) => res.info.avgRating >= 4.3
              );
              setFilteredRestaurants(filteredList);
         
            }}
          >
            Top Rated Restaurants
          </button>
        </div>
        <div className="search m-1 p-3">
          <input
            type="text"
            className="px-4 py-2 border rounded-lg border-solid border-black"
            value={searchText}
            onChange={(e) => {
              setSearchText(e.target.value);
            }}
          />
          <button
            className="px-4 py-2 bg-green-100 m-4 rounded-lg border border-emerald-950"
            onClick={() => {
              console.log(searchText);
              const filteredRestaurant = listOfRestaurants.filter((res) =>
                res?.info?.name
                  ?.toLowerCase()
                  .includes(searchText.toLowerCase())
              );

              setFilteredRestaurants(filteredRestaurant);
            }}
          >
            Search
          </button>
        </div>
      </div>
      <div className="flex flex-wrap m-3 p-3 justify-start">
        {filteredRestaurants.map((restaurants) => (
          <Link
            key={restaurants.info.id}
            to={"/restaurant/" + restaurants.info.id}
          >
            {
              /* if the restaurant is promoted then add a promoted label to it */
              restaurants.info.veg ? (
                <RestaurantCardPromoted resData={restaurants}/>
              ) : (
                <RestaurantCard resData={restaurants} />
              )
            }
          </Link>
        ))}
      </div>
    </div>
  );
};
export default Body;
`

**Reply by alex-krasikau** [reply]
Hey there!

Which file extension do you use for React.js files? Is it `.js` or `.jsx`?

On a similar note, why aren't you open to checking out other frameworks?

**Reply by princepalewar12** [reply]
I am using .js extension in my react project.
actually i am almost done with this project just want to give it some styling and that's it. That's why i don't want to use other framework and i also want to get some experience with taiwindcss.

**Comment by alex-krasikau** [comment]
I don't have a clue yet. Could you please check out the following guide and try to replicate the configuration? See if that solves the issue. Here's the link: [https://dev.to/matthewsaintbull/parcel-react-tailwind-to-the-moon-37i3](https://dev.to/matthewsaintbull/parcel-react-tailwind-to-the-moon-37i3).

**Reply by princepalewar12** [reply]
yeah sure thanks

**Reply by sammarabbasdev** [reply]
Thanks
